src/storage.js

```diff
@@ -0,0 +1,129 @@
+'use strict'
+
+const {
+  DynamoDBClient,
+  GetItemCommand,
+  PutItemCommand,
+  DeleteItemCommand,
+  UpdateItemCommand
+} = require('@aws-sdk/client-dynamodb')
+const { SQSClient, SendMessageCommand } = require('@aws-sdk/client-sqs')
+const { NodeHttpHandler } = require('@aws-sdk/node-http-handler')
+const {
+  marshall: serializeDynamoItem,
+  unmarshall: deserializeDynamoItem,
+  convertToAttr
+} = require('@aws-sdk/util-dynamodb')
+const { Agent } = require('https')
+const { base58btc: base58 } = require('multiformats/bases/base58')
+
+const { logger, serializeError } = require('./logging')
+
+const agent = new Agent({ keepAlive: true, keepAliveMsecs: 60000 })
+
+const dynamoClient = new DynamoDBClient({
+  requestHandler: new NodeHttpHandler({ httpsAgent: agent })
+})
+
+const sqsClient = new SQSClient({
+  requestHandler: new NodeHttpHandler({ httpsAgent: agent })
+})
+
+function cidToKey(cid) {
+  return base58.encode(cid.multihash.bytes)
+}
+
+async function readDynamoItem(table, keyName, keyValue) {
+  try {
+    const record = await dynamoClient.send(
+      new GetItemCommand({ TableName: table, Key: serializeDynamoItem({ [keyName]: keyValue }) })
+    )
+
+    if (!record.Item) {
+      return null
+    }
+
+    return deserializeDynamoItem(record.Item)
+  } catch (e) {
+    logger.error(`Cannot get item from DynamoDB table ${table}: ${serializeError(e)}`)
+    throw e
+  }
+}
+
+async function writeDynamoItem(create, table, keyName, keyValue, data, conditions = {}) {
+  let command
+
+  if (create) {
+    command = new PutItemCommand({
+      TableName: table,
+      Item: serializeDynamoItem({ [keyName]: keyValue, ...data }, { removeUndefinedValues: true })
+    })
+  } else {
+    const update = []
+    const values = {}
+    const condition = []
+
+    for (const [key, value] of Object.entries(data)) {
```

Comment:
we shouldn’t need any of this indexed right?

we don’t need link indexing in this version so we can avoid decoding the block at all and storing any of the decoded data. should speed things up a lot.

Reply:
Yes, exactly. But as I disabled decoding before this comes for free.